### ISSUE TITLE
client["/endpoint"].GET style call pattern to enable go-to-definition

### ISSUE THREAD

**Issue by tonyxiao** [openapi-ts/openapi-typescript#1526]
**Description**

Navigating from call site to docs (aka generated typing) today is quite inconvenient as vscode is not able to navigate from the path argument parameter to the type def. I would love to be able to use cmd+click to be able to go straight to the generated types for the specific endpoint. This is similar to how trpc would work 

**Proposal**

**Before**
```ts
client.GET('/endpoint', { params: {} })
```

**After**
```ts
client['/endpoint'].GET({ params: {} })
```

This way you can cmd+click on the `/endpoint` part in VSCode to go straight to defintion

We can keep both call styles too (just like TRPC) for backward compat. 

**Checklist**

- [x] I’m willing to open a PR for this (see [CONTRIBUTING.md](https://github.com/drwpow/openapi-typescript/blob/main/packages/openapi-fetch/CONTRIBUTING.md))

**Comment by drwpow** [comment]
I’m all for better Intellisence and VS Code integration, but how would the runtime work? Would `client` have to be a getter method? And would this be able to use `openapi-typescript`’s types as-generated? I’d be curious to see a PoC on how this could work from both a runtime and type inference side, even with a simple schema.

**Comment by tonyxiao** [comment]
Sure. This it how it is used. 

https://github.com/trpc/trpc/blob/44a1b58a3f21dae230cf290a71d93614d469e63b/examples/next-minimal-starter/src/pages/index.tsx#L8

This is how trpc implements it

https://github.com/trpc/trpc/blob/44a1b58a3f21dae230cf290a71d93614d469e63b/packages/client/src/createTRPCClient.ts#L126-L145

**Comment by gzm0** [comment]
This is fixed in #1791 / openapi-fetch 0.11.0.
